rplugin/python3/deoplete/sources/tabnine.py

```diff
@@ -154,19 +162,28 @@ def _restart(self):
         if self._proc is not None:
             self._proc.terminate()
             self._proc = None
+            self._selector.close()
+            self._selector = None
         binary_dir = os.path.join(self._install_dir, 'binaries')
         path = get_tabnine_path(binary_dir)
         if path is None:
             self.print_error('no TabNine binary found')
             return
         self._proc = subprocess.Popen(
-            [path, '--client', 'sublime', '--log-file-path',
-             os.path.join(self._install_dir, 'tabnine.log')],
+            [
+                path,
+                '--client', 'sublime',
+                '--log-file-path', os.path.join(self._install_dir, 'tabnine.log'),
+                '--idle_suicide_seconds', '3600',
```

Comment:
Unfortunately, this is currently has no effect on the operation of the binary.

Reply:
Thats a shame, as the binary tends to bloat over time. Last week it gorged itself on all my RAM, all 128 GB of it.

Reply:
We have code inside the binary that should check such things. If you come across such bug again please open us an issue or contact support. We would love to resolve it!